src/main/java/edu/ucsb/cs156/happiercows/jobs/UpdateCowHealthJob.java

```diff
@@ -43,12 +47,22 @@ public void accept(JobContext ctx) throws Exception {
                 Integer totalCows = commonsRepository.getNumCows(commons.getId()).orElseThrow(()->new RuntimeException("Error calling getNumCows(" + commons.getId() + ")"));
 
                 for (UserCommons userCommons : allUserCommons) {
+                    double totalHealths = 0d;
                     User user = userRepository.findById(userCommons.getUserId()).orElseThrow(()->new RuntimeException("Error calling userRepository.findById(" + userCommons.getUserId() + ")"));
                     ctx.log("User: " + user.getFullName() + ", numCows: " + userCommons.getNumOfCows() + ", cowHealth: " + userCommons.getCowHealth());
-                        double newCowHealth = calculateNewCowHealth(userCommons.getCowHealth(), userCommons.getNumOfCows(), totalCows, carryingCapacity, degradationRate);
-                        ctx.log("old cow health: " + userCommons.getCowHealth() + ", new cow health: " + newCowHealth);
-                        userCommons.setCowHealth(newCowHealth);
-                        ctx.log("Cow health has been updated!");
+                    for(CowLot cowLot: cowLotRepository.findAllByUserCommonsId(userCommons.getId())){
+                        double newCowHealth = calculateNewCowHealth(cowLot.getHealth(), userCommons.getNumOfCows(), totalCows, carryingCapacity, degradationRate);
+                        ctx.log("old cow health: " + cowLot.getHealth() + ", new cow health: " + newCowHealth);
+                        cowLot.setHealth(newCowHealth);
+                        if(newCowHealth > 0){
+                            cowLotRepository.save(cowLot);
+                        } else {
+                            cowLotRepository.delete(cowLot);
+                            userCommons.setNumOfCows(userCommons.getNumOfCows()-cowLot.getNumCows());
+                        }
+                        totalHealths += newCowHealth * cowLot.getNumCows();
+                    }
+                    userCommons.setCowHealth(totalHealths / userCommons.getNumOfCows());
```

Comment:
Although it may seems inefficient to have extra variables that are only used once, it can be a way to document things for the reader.  Here's a good example:

```suggestion
                    double averageCowHealthThisCommons = sumNewHealthValueAllCowsThisCommons / userCommons.getNumOfCows();
                    userCommons.setCowHealth(averageCowHealthThisCommons);
```

Reply:
Fixed!